src/main/python/interfaces/app/appSidebarUI.py

```diff
@@ -0,0 +1,98 @@
+from fbs_runtime.application_context.PyQt5 import ApplicationContext
+
+from PyQt5.QtCore import Qt
+
+from ..widgets import Frame, VerticalLayout, VerticalSpacer, Label, Button
+
+
+class AppSidebarUI(Frame):
+
+    dashboard: Button = None
+    resources: Button = None
+    jobs: Button = None
+    settings: Button = None
+
+    def __init__(
+        self, parent, cxt: ApplicationContext, width: int, height: int, *args, **kwargs
+    ):
+        super(AppSidebarUI, self).__init__(parent, name="sidebar")
+
+        self.cxt = cxt
+
+        self._init_ui(width, height)
+
+        self.setStyleSheet(self.cxt.app_style)
+
+    def reload_styleSheet(self):
+        self.setStyleSheet(self.cxt.app_style)
+
+    def on_dashboard_clicked(self):
+        self.reset()
+        self.dashboard.setObjectName("sidebar_button_active")
+        self.reload_styleSheet()
+
+    def on_resources_clicked(self):
+        self.reset()
+        self.resources.setObjectName("sidebar_button_active")
+        self.reload_styleSheet()
+
+    def on_jobs_clicked(self):
+        self.reset()
+        self.jobs.setObjectName("sidebar_button_active")
+        self.reload_styleSheet()
+
+    def on_settings_clicked(self):
+        self.reset()
+        self.settings.setObjectName("sidebar_button_active")
+        self.reload_styleSheet()
+
+    def reset(self):
+        # Set all widget to base styles and let clicked callback override them
+        self.dashboard.setObjectName("sidebar_button")
+        self.resources.setObjectName("sidebar_button")
+        self.jobs.setObjectName("sidebar_button")
+        self.settings.setObjectName("sidebar_button")
+
+    def _init_ui(self, width: int, height: int):
+
+        self.setFixedSize(width, height)
+
+        window_layout = VerticalLayout(self)
+
+        # title
+
```

Comment:
Try to clean up a bit. It was just something when Fenting working the code, she prefer to leave a comment wheat the code need to be filled in. By the way, it will be nice if we can keep some of them. While there's long function for a widget, there's easy to get lose.

Reply:
ah I see.  these comments should definitely be removed when the function is implemented.  Maybe mark with a `TODO` to make tracking easier? @fennnn